components/brave_wallet/renderer/brave_wallet_js_handler.cc

```diff
@@ -393,6 +395,24 @@ v8::Local<v8::Promise> BraveWalletJSHandler::Request(
         formed_input,
         base::BindOnce(&OnAddEthereumChain, std::move(promise_resolver),
                        isolate, std::move(context_old)));
+  } else if (method && *method == kEthSendTransaction) {
+    std::string formed_input;
+    // Hardcode id to 1 as it is unused
+    ALLOW_UNUSED_LOCAL(out_dict->SetIntPath("id", kRequestId));
+    ALLOW_UNUSED_LOCAL(out_dict->SetStringPath("jsonrpc", kRequestJsonRPC));
+    if (!base::JSONWriter::Write(*out_dict, &formed_input))
+      return v8::Local<v8::Promise>();
+
+    std::string from;
+    auto tx_data = ParseEthSendTransactionParams(formed_input, &from);
+    if (!tx_data)
+      tx_data = mojom::TxData::New();
+
+    brave_wallet_provider_->AddUnapprovedTransaction(
+        std::move(tx_data), from,
+        base::BindOnce(&BraveWalletJSHandler::OnAddUnapprovedTransaction,
+                       base::Unretained(this), std::move(promise_resolver),
```

Comment:
It seems like we can just make `OnAddUnapprovedTransaction` a free function in anonymous namespace and remove `Unretained(this)`.

Reply:
Fixed in d8c9c09e62fadd786f09bfdf932a2573f3dd5358